GW2EIEvtcParser/EncounterLogic/Strikes/EndOfDragons/AetherbladeHideout.cs

```diff
@@ -268,5 +277,87 @@ internal override FightData.EncounterMode GetEncounterMode(CombatData combatData
             }
             return maiTrin.GetHealth(combatData) > 8e6 ? FightData.EncounterMode.CM : FightData.EncounterMode.Normal;
         }
+
+        protected override void SetInstanceBuffs(ParsedEvtcLog log)
+        {
+            base.SetInstanceBuffs(log);
+            IReadOnlyList<AbstractBuffEvent> triangulation = log.CombatData.GetBuffData(AchievementEligibilityTriangulation);
+            bool hasTriangulationBeenAdded = false;
+            
+            if(log.FightData.Success)
+            {
+                if (triangulation.Any())
+                {
+                    foreach (Player p in log.PlayerList)
+                    {
+                        if (p.HasBuff(log, AchievementEligibilityTriangulation, log.FightData.FightEnd - ServerDelayConstant))
+                        {
+                            InstanceBuffs.Add((log.Buffs.BuffsByIds[AchievementEligibilityTriangulation], 1));
+                            hasTriangulationBeenAdded = true;
+                            break;
+                        }
+                    }
+                }
+                if (!hasTriangulationBeenAdded && CustomCheckTriangulationEligibility(log))
+                {
+                    InstanceBuffs.Add((log.Buffs.BuffsByIds[AchievementEligibilityTriangulation], 1));
+                }
+            }
+        }
+
+        private static bool CustomCheckTriangulationEligibility(ParsedEvtcLog log)
+        {
+            IReadOnlyList<long> beamsBuffs = new List<long>() { MaiTrinCMBeamsTargetBlue, MaiTrinCMBeamsTargetGreen, MaiTrinCMBeamsTargetRed };
+            var beamsSegments = new List<Segment>();
+            var bombInvulnSegments = new List<Segment>();
+
+            foreach (AgentItem player in log.PlayerAgents)
+            {
+                IReadOnlyDictionary<long, BuffsGraphModel> bgms = log.FindActor(player).GetBuffGraphs(log);
+                foreach (long buff in beamsBuffs)
+                {
+                    beamsSegments = GetBuffSegments(bgms, buff, beamsSegments).OrderBy(x => x.Start).ToList();
+                }
+            }
+
+            foreach (AgentItem agent in log.AgentData.GetNPCsByID(ArcDPSEnums.TrashID.FerrousBomb))
+            {
+                IReadOnlyDictionary<long, BuffsGraphModel> bgms = log.FindActor(agent).GetBuffGraphs(log);
+                bombInvulnSegments = GetBuffSegments(bgms, FailSafeActivated, bombInvulnSegments).OrderBy(x => x.Start).ToList();
+            }
+
+            int counter = 0;
+
+            // For each segment where a bomb is invulnerable, check if it has started between the assignment and loss of a beam effect on a player
+            // If the counter is == 8, it means every possible combination check has been met, meaning it's eligible for the achievement.
```

Comment:
Why 8?

Reply:
It checks 2 players buffs for each bomb invuln, for 4 bombs, so a total of 8 combinations where even 1 of them being different would result into no achievement. Otherwise i can trim 2 of the 4 invuln buffs (the one with a higher start time), resulting in a counter of 4

Reply:
No, that's okay, perhaps put that down also in comment, I just needed to understand the logic.